webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/index.js

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { initialUpdate, updateSelected } from './TypeAheadSelectActions';
+import { selectOptions, selectSelected } from './TypeAheadSelectSelectors';
+import reducer from './TypeAheadSelectReducer';
+import TypeAheadSelect from './TypeAheadSelect';
+
+const ConnectedTypeAheadSelect = ({
+  id,
+  options,
+  selected,
+  allowNew,
+  multiple,
+  placeholder,
+  defaultInputValue,
+  clearButton,
+  inputProps,
+}) => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(initialUpdate(options, selected, id));
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Regarding the exhaustive-deps eslint rule -- Instead of disabling the eslint rule, I think the "correct" way to avoid that Hooks error would be
* Add a selector `doesTypeAheadSelectExist` which checks for existence by ID
* Add a condition so it doesn't dispatch the action if it already exists
* Add `options`, `selected`, and `id` to the dependency array

IMO this is the hardest part of useEffect

Reply:
I see your point and I think I like it - trying it out

Reply:
Good approach, I made the change